src/views/Store.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { useReducer } from 'react';
+import { ToCart } from '../components/ToCart';
+import { ShoppingCart } from '../components/ShoppingCart';
+
+const someItems = [
+    { id: 0, text: 'Potion', done: false }, 
+    { id: 1, text: 'Ether', done: false }, 
+    { id: 2, text: 'Bomb', done: false } 
+]
+
+
+const addUpdate = (items, action) => {
+    switch (action.type) {
+        case 'added' : {
+            return [...items, {
+                id: action.id,
+                text: action.text,
+                done: false
+            }]
+         }
+         case 'updated': {
+            return items.map((item) => {
+                if (item.id === action.task.id) {
+                    return action.task
+                }
+                return item
+            })
+        }
+        case 'deleted': {
+            return items.filter((item) => 
+            item.id !== action.id)
+        }
+        default: {
+          return someItems
+        }
+    }
+}
+const add = 3;
+
+export const Store = () => {
+    const [items, dispatch] = useReducer(addUpdate, someItems)
+
+    const addItem = (text) => {
+        dispatch({
+            type: 'added',
+            id: add + 1, 
```

Comment:
Good idea with adding 1 to the id based on a preset value of your starting items! 

Another shorter way of writing that would be to call the `.length` property of your `someItems` array directly. 

`id: someItems.length + 1,`